gobblin-utility/src/main/java/org/apache/gobblin/util/AvroUtils.java

```diff
@@ -612,6 +614,127 @@ public static Schema switchName(Schema schema, String newName) {
   }
 
   /**
+   * Copies the input {@link org.apache.avro.Schema} but changes the schema namespace.
+   * @param schema {@link org.apache.avro.Schema} to copy.
+   * @param namespaceOverride namespace for the copied {@link org.apache.avro.Schema}.
+   * @return A {@link org.apache.avro.Schema} that is a copy of schema, but has the new namespace.
+   */
+  public static Schema switchNamespace(Schema schema, Map<String, String> namespaceOverride) {
+    Schema newSchema;
+    String newNamespace = StringUtils.EMPTY;
+
+    // Process all Schema Types
+    // (Primitives are simply cloned)
+    switch (schema.getType()) {
+      case ARRAY:
+        newSchema = Schema.createArray(switchNamespace(schema.getElementType(), namespaceOverride));
+        break;
+      case BOOLEAN:
```

Comment:
A more compact organization would be to group the primitive types together at the beginning with fall through to avoid repeating newSchema = Schema.create(schema.getType()).